pageserver/src/tenant/timeline.rs

```diff
@@ -3542,25 +3544,23 @@ impl Timeline {
         // This iterator walks through all key-value pairs from all the layers
         // we're compacting, in key, LSN order.
         let all_values_iter = itertools::process_results(
-            deltas_to_compact.iter().map(|l| l.iter(ctx)),
+            deltas_to_compact
+                .iter()
+                .map(|l| -> Result<_> { Ok(l.load_val_refs(ctx)?.into_iter()) }),
             |iter_iter| {
                 iter_iter.kmerge_by(|a, b| {
-                    if let Ok((a_key, a_lsn, _)) = a {
-                        if let Ok((b_key, b_lsn, _)) = b {
-                            (a_key, a_lsn) < (b_key, b_lsn)
-                        } else {
-                            false
-                        }
-                    } else {
-                        true
-                    }
+                    let (a_key, a_lsn, _) = a;
```

Comment:
copying the key might be costly as it is 48B in size IIRC. probably we can use `let (ref a_key, a_lsn, _) = a` here?

Reply:
The `ref` is automatically inserted by the compiler since [match ergonomics](https://rust-lang.github.io/rfcs/2005-match-ergonomics.html) have been added in 2017/2018. If you look at the type hints in the IDE you also see that no copy is happening:


![Screenshot_20230731_172307](https://github.com/neondatabase/neon/assets/19521842/d3fe5428-18c3-46c6-96a4-a4d7aae900d0)

Reply:
that's cool, thanks for the explanation!